pkg/kubelet/volumemanager/volume_manager.go

```diff
@@ -322,6 +323,16 @@ func (vm *volumeManager) GetVolumesInUse() []v1.UniqueVolumeName {
 	return volumesToReportInUse
 }
 
+func (vm *volumeManager) verifyIfInUse(volume cache.AttachedVolume) bool {
```

Comment:
I think we don't need to verifyIfInUse because it might cause an race condition again. E.g., when you check IsOperationPending, it just finishes. We can just use allAttachedVolumes and put them VolumeInUse. After pod is deleted, reconciler has the logic to remove volume from attachedVolumes list if it is not mounted. Then kubelet volume manager will update the volume is no longer in use.

Reply:
but I agree, I think it is safer to just add all volumes. I have updated the PR.

Reply:
right. I just think attachedVolume could cover the missing volumes without any potential race condition problem. So checking whether it is pending or not is not very necessary and might cause the race condition I just mentioned although it should be very rare. As long as the volume is in attachedVolumes list, the volume might be mounted any moment. Checking globallymounted and pending operation is asynchronized with the mounting operation, so there will be race condition there.